sig-robotics/proposal/Definition of cloud robot CRD and Operator/Definition of cloud robot CRD and Operator.md

```diff
@@ -0,0 +1,157 @@
+|     registed_robots     | string[] | Keep a list of registered nodes, including nodes name and a list of node devices |
+
+## Controller design
+
+The Controller is mainly used to monitor the newly added custom resources in the cluster, or the changes of existing resources, and execute corresponding logic based on these informations.
+
+For *taskController*
+
+1. When a new task resource is created in the cluster, taskController should monitor this change
+2. The taskController match the task to a node that can execute the task
+3. Then the Controller should modify the task_info field of the node
+
+For the *robotControlelr*
+
+1. The robotController monitors the changes of the robot resources
+2. According to the changes, the Controller send the corresponding command to the node
+
+For the *registrationController*
+
+1. Listen for registration messages from robot nodes, and monitor the status of node at the same time
+2. Creating the robot CR if it receives the registration message and maintain the informations of robotSync CR. If the node goes offline, set the DeletionTimeStamp for the corresponding robot CR
+
+The processing logic of each Controller is shown in the figure below:
+
+<img src="./images/controllers.png" alt="controllers" style="zoom:80%;" />
+
+### Robot Registration
+
+<img src="./images/registration.png" alt="registration" style="zoom:90%;" />
+
+1. Registration Controller judges whether the node has been registered by comparing the number of nodes and robot CR
+2. If a node is not registered, send a message to the node to remind the registration
+3. When receiving a registration request, on the one hand, create a robot CR, and on the other hand, fill in the information in the registration request to robotSync
+
+### Use case
+
+Take the fleet scheduling system as an example:
+
+![fleet_scheduling](./images/fleet_scheduling.png)
+
+Fleet scheduling can be divided into four stages
+
+1. vehicle registration:
+   - The registrationController checks the number of registered robots in the robotSync CR(create it if not exist) and the number of nodes ,  if they are inconsistent, it sends a message to the unregistered node to remind the registration
+   - After the Registration Controller receives the registration request, it creates a robot CR and updates the registration information to the robotSync CR at the same time
+2. task release:
+   - The user creates a task CR on the cloud according to the specific task process
+3. task assignment:
+   - The taskController monitors the creation of the task CR and obtains the task content then selects a suitable robot from the registered robot list in robotSync to assign this task. It also modifies the task_info field in the corresponding robot CR
+4. task execution:
+   - The robotController listens to changes in the fields of the robot CR and obtains task-related information then it sends action messages to the corresponding nodes
```

Comment:
When the robot network is unstable, how can we ensure that the action messages sent by robotController to robots are not lost?